### PR TITLE
GEOMETRY-63: Fix some SonarQube smells

### PR DESCRIPTION
**Pull request by singhbaljit**
- Mostly addresses JUnit assertions
https://sonarcloud.io/dashboard?id=commons-geometry

### CODE REVIEW

**Comment by darkma773r** [comment]
Apologies, Baljit! I didn't realize that you had a pull request open. I'll take a look now.

File: commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java [hunk 1]
```diff
@@ -199,13 +199,13 @@ public void testEquals() {
         EpsilonDoublePrecisionContext c = new EpsilonDoublePrecisionContext(1e-6);
 
         // act/assert
-        Assert.assertFalse(a.equals(null));
-        Assert.assertFalse(a.equals(new Object()));
-        Assert.assertFalse(a.equals(b));
-        Assert.assertFalse(b.equals(a));
+        Assert.assertNotEquals(null, a);
+        Assert.assertNotEquals(new Object(), a);
```
Comment (darkma773r) [inline commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java:203]: JUnit applies its own null checks in the `assertEquals` and `assertNotEquals` methods, meaning that we end up missing those branches in our code. This seems to be why the code coverage decreased. Let's revert these changes on this and all of the other `equals()` method tests and see if the code coverage goes back up.

Comment (singhbaljit) [reply commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java:203]: Undid the tests for null checks.

**Comment by singhbaljit** [comment]
There is also https://jqno.nl/equalsverifier 😄
